sympy/functions/special/delta_functions.py

```diff
@@ -193,8 +193,16 @@ def eval(cls, arg, k=0):
             return S.NaN
         if arg.is_nonzero:
             return S.Zero
+        c, nc = arg.args_cnc()
+        if c and c[0] == -1:
+            # keep this fast and simple instead of using
+            # could_extract_minus_sign
+            return cls(-arg, k) if k else cls(-arg)
```

Comment:
The derivative of order `k` has the same parity as `k`. The first derivative of an even function is odd, the second derivative is even, etc.